File: 151-reverse-words-in-a-string/reverse-words-in-a-string.cpp
#include<sstream>
#include<string>
#include<stack>
class Solution {
public:
    string reverseWords(string s) {
        stack<string> st;
        stringstream ss(s);
        string word;
        while(ss >> word){
           st.push(word);
        }
        string ans = "";
        while(!st.empty()){
            ans+=st.top();
            st.pop();
            if(!st.empty()){
                ans += " ";
            }
        }
        return ans;
    }
};

File: 151-reverse-words-in-a-string/reverse-words-in-a-string.java
class Solution {
    public String reverseWords(String s) {
        String ans = "";
        int n = s.length();
        int j = 0;
        while(j<n){
            while(j<n && s.charAt(j) == ' ') j++;
            String temp = "";
            while(j<n && s.charAt(j) != ' ') {
                temp = temp + s.charAt(j);
                j++;
            }
            ans  = " " + temp  +ans;
        }
        return ans.trim();
    }
}